File: Georechner_C++/grundlagen/erstegrund.cpp
#include "mainwindow.h"
#include <iostream>

float* umrechnen_koordinaten(float& y1, float& x1, float& s12, float& t12){
    float sin_t12 = sin(gon2rad(t12));
    float cos_t12 = cos(gon2rad(t12));

    float y2 = y1+s12*sin_t12;
    float x2 = x1+s12*cos_t12;

    static float arr_y2_x2[] = {y2, x2};
    return arr_y2_x2;
}

float umrechnen_koordinatenunterschiedey(float& s, float& t){
    float sin_t = sin(gon2rad(t));
    float dy = floor(s*sin_t + 0.5);
    return dy;
}

float umrechnen_koordinatenunterschiedex(float& s, float& t){
    float cos_t = cos(gon2rad(t));
    float dx = floor(s*cos_t +0.5);
    return dx;
}

File: Python/grundlagen/erstegrund.py
from math import *
from grundlagen import winkel

def umrechnen_koordinaten(y1,x1,s12,t12):
    sin_t12=sin(winkel.gon2rad(t12))
    cos_t12=cos(winkel.gon2rad(t12))

    y2=y1+s12*sin_t12
    x2=x1+s12*cos_t12

    return y2,x2

def umrechnen_koordinatenunterschiedey(s, t):
    sin_t = sin(winkel.gon2rad(t))
    dy = round(s*sin_t,2)
    return dy

def umrechnen_koordinatenunterschiedex(s, t):
    cos_t = cos(winkel.gon2rad(t))
    dx = round(s*cos_t,2)
    return dx

